### PR TITLE
Fix erroneously created http url directory path for monitoring

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

Fixes erroneously created http url directory path for monitoring when the agent runs. 
Added a check if the url is passed in the path.

```
➜ sudo ls -la /Library/Elastic/Agent/http\:/localhost\:6791
total 0
drwxr-x---  2 root  wheel  64 Nov 28 11:12 .
drwxr-x---  3 root  wheel  96 Nov 28 11:12 ..
```

## Why is it important?

No erroneous directories should be created during the agent execution.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## How to test this PR locally

1. Install the agent
2. Make sure the agent create "http" directory, something like: ``` /Library/Elastic/Agent/http\:/localhost\:6791``` 

- Closes https://github.com/elastic/elastic-agent/issues/1806

### CODE REVIEW

File: internal/pkg/agent/application/monitoring/server.go [hunk 1]
```diff
@@ -75,7 +76,7 @@ func exposeMetricsEndpoint(
 }
 
 func createAgentMonitoringDrop(drop string) error {
-	if drop == "" || runtime.GOOS == "windows" {
+	if drop == "" || runtime.GOOS == "windows" || isHttpUrl(drop) {
```
Comment (cmacknz) [inline internal/pkg/agent/application/monitoring/server.go:79]: This looks like it fixes it, but why is an HTTP URL even being passed here?

Comment (aleksmaus) [reply internal/pkg/agent/application/monitoring/server.go:79]: @blakerouse do you know the history on this? seems like the monitoring changed from http to the local socked between V1 and V2?

**Review by cmacknz — APPROVED** [review]
Approving so we don't have a random http URL in our directory tree on installs, which looks a bit embarrassing. We can figure out where this is coming from separately.

**Pull request merged**